go/citogo/main.go

```diff
@@ -148,106 +155,104 @@ func (r *runner) flushTestLogsToTemp(logName string, log bytes.Buffer) (string,
 	return fmt.Sprintf("see log: %s", tmpfile.Name()), nil
 }
 
-func (r *runner) reportFlake(test string, logs string) {
-	hook := os.Getenv("CITOGO_FLAKE_WEBHOOK")
-	if hook == "" {
+func (r *runner) report(result TestResult) {
+	functionName := "report-citogo"
+
+	b, err := json.Marshal(result)
+	if err != nil {
+		logError("error marshalling result: %w", err)
 		return
 	}
 
-	r.doHook(hook, test, logs, "❄️")
-}
-
-func (r *runner) doHook(hook string, test string, logs string, emoji string) {
-	hook += url.QueryEscape(fmt.Sprintf("%s _client_ %s-%s %s *%s* %s [%s]", emoji, r.opts.Branch, r.opts.BuildID, r.opts.Prefix, test, logs, r.opts.BuildURL))
-	_, err := http.Get(hook)
+	req := r.lambdaCli.InvokeRequest(&lambda.InvokeInput{
+		FunctionName: &functionName,
+		Payload:      b,
+	})
+	_, err = req.Send(context.TODO())
 	if err != nil {
 		logError("error reporting flake: %s", err.Error())
 	}
 }
 
-type outcome string
+type Outcome string
 
 const (
-	success outcome = "success"
-	flake   outcome = "flake"
-	fail    outcome = "fail"
+	OutcomeSuccess Outcome = "success"
+	OutcomeFlake   Outcome = "flake"
+	OutcomeFail    Outcome = "fail"
 )
 
-func (o outcome) abbrv() string {
+func (o Outcome) Abbrv() string {
 	switch o {
-	case success:
+	case OutcomeSuccess:
 		return "PASS"
-	case flake:
+	case OutcomeFlake:
 		return "FLK?"
-	case fail:
+	case OutcomeFail:
 		return "FAIL"
 	default:
 		return "????"
 	}
 }
 
-func (r *runner) reportTestOutcome(outcome outcome, test string, where string) {
-	fmt.Printf("%s: %s", outcome.abbrv(), test)
-	if where != "" {
-		fmt.Printf(" %s", where)
+func (r *runner) runTest(test string) error {
+	outcome, where, err := r.runTestOnce(test, false /* isRerun */)
+	if err != nil {
+		return err
 	}
-	fmt.Printf("\n")
-
-	if outcome != fail || r.opts.Branch != "master" {
-		return
+	if outcome == OutcomeSuccess {
+		return nil
 	}
-	hook := os.Getenv("CITOGO_MASTER_FAIL_WEBHOOK")
-	if hook == "" {
-		return
+	if len(r.flakes) >= r.opts.Flakes {
```

Comment:
I'm not sure exactly what the purpose of this config value is. Probably not using it right either.

Reply:
it's the total number of flakes a test run is allowed before it just fails out

Reply:
what about the test for `err == errTestFailed`?    Don't want to rerun the test if `runTestOnce` failed for a different reason